modules/optimization/doc/content/modules/optimization/examples/shapeOpt_Annulus.md

```diff
@@ -0,0 +1,122 @@
+# Constrained Optimization Example: Shape Optimization
+
+## Background
+
+The MOOSE Optimization module provides a flexible framework for solving
+optimization problems in MOOSE. This page will demonstrate constrained shape
+optimization of an annulus while satisfying an equality volume constraint.
```

Comment:
It would be great if you could add a few sentences on what shape optimization is in general terms. Adding a citation would help a lot too.